src/sentry/search/snuba/backend.py

```diff
@@ -147,16 +149,27 @@ def owner_filter(owner, projects):
                 is_active=True,
             ).values("team")
         )
-        relevant_owners = GroupOwner.objects.filter(
+        owned_by_me = GroupOwner.objects.filter(
             project_id__in=project_ids, organization_id=organization_id
-        )
-        filter_query = Q(team__in=teams) | Q(user=owner)
-        return Q(
-            id__in=relevant_owners.filter(filter_query)
+        ).filter(Q(team__in=teams) | Q(user=owner))
+        ids_assigned_to_someone_else = (
+            GroupAssignee.objects.filter(
```

Comment:
Do you think we'd ever merge the owner and assignee models? It seems like they have a lot of overlap and that one of the potential owners could be flagged as the 'assigned owner'. Having the models joined could help simplify these queries but I've not thought through the other implications of that change.

Reply:
That's an interesting thought. I'll definitely think about that and the implications it could have and bring it up with Dan (and the team as a whole). The idea of adding a third type to the GroupOwner choices like "Assigned Owner" definitely sits well with me on a lot of levels, but we'd have to change all of the code that references GroupAssignee and likely some GroupOwner code too.

In the short-term, assuming this doesn't have other issues that shouldn't be merged, I think we'll implement it like this and look towards possible improvements it in the future (this is being treated as a fairly high priority task as it is blocking inbox adoption).

Reply:
Refining the tables can totally be done 'later'.